File: back_end/go/5.go
package main

func calc(f func(int, int) int, a int, b int) int {
	result := f(a, b)
	return result
}

func main() {
	sum := func(n ...int) int{
		s := 0
		for _, i := range n{
			s += i
		}
		return s
	}
	result := sum(1, 2, 3, 4, 5)
	println(result)

	add := func(i int, j int) int {
		return i + j
	}

	// add 함수 전달
	r1 := calc(add, 10, 20)
	println(r1)

	// 직접 첫번째 파라미터에 익명함수를 정의함
	r2 := calc(func(x int, y int) int { return x - y }, 10, 20)
	println(r2)
}

File: algorithm/LC/5.py
# -*- coding: utf-8 -*-
class Solution:
    def longestPalindrome(self, s: str) -> str:

        if len(s) < 2 or s == s[::-1]:
            return s

        result = ''

        def long(start, end):
            while start >= 0 and end <= len(s) and s[start] == s[end-1]:
                start -= 1
                end += 1
            return s[start+1:end-1]

        for i in range(len(s)-1):
            result = max(result, long(i, i+1), long(i, i+2), key=len)

        return result


s = Solution()
# assert s.longestPalindrome("babad") == "bab"
assert s.longestPalindrome("abb") == "bb"
